### PR TITLE
feat: build: Check for manifest CID while loading bundle

### PR DESCRIPTION
**Pull request by geoff-vball**
## Related Issues
Closes https://github.com/filecoin-project/lotus/issues/8682

## Checklist

Before you mark the PR ready for review, please make sure that:
- [x] All commits have a clear commit message.
- [ ] The PR title is in the form of of `<PR type>: <area>: <change being made>`
    - example: ` fix: mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_,_perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_, _deps_
- [x] This PR has tests for new functionality or change in behaviour
- [x] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in https://lotus.filecoin.io or [Discussion Tutorials.](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [x] CI is green

### CODE REVIEW

File: cmd/lotus-shed/cid.go [hunk 1]
```diff
@@ -83,3 +91,49 @@ var cidIdCmd = &cli.Command{
...
+	Action: func(cctx *cli.Context) error {
+		ctx := context.TODO()
+
+		cf := cctx.Args().Get(0)
+		f, err := os.OpenFile(cf, os.O_RDONLY, 0664)
+		if err != nil {
+			return xerrors.Errorf("opening the car file: %w", err)
+		}
+
+		bs := modules.MemoryBlockstore()
```
Comment (arajasek) [inline cmd/lotus-shed/cid.go:108]:
```suggestion
		bs := blockstore.NewMemory()
```

**Review by arajasek — APPROVED** [review]
Build sad, but LGTM.

**Review by vyzo — COMMENTED** [review]
This was the wrong way yo do it.
The right way yo do it would be to have it in the bundles.toml, optionally for each neteork/telease that supports it, and check on load if it wa set.

File: cmd/lotus-shed/cid.go [hunk 2]
```diff
@@ -83,3 +91,41 @@ var cidIdCmd = &cli.Command{
...
+		cf := cctx.Args().Get(0)
+		f, err := os.OpenFile(cf, os.O_RDONLY, 0664)
+		if err != nil {
+			return xerrors.Errorf("opening the car file: %w", err)
+		}
+
+		bs := blockstore.NewMemory()
+		if err != nil {
+			return err
+		}
```
Comment (travisperson) [inline cmd/lotus-shed/cid.go:111]: There is no error here

File: cmd/lotus-shed/cid.go [hunk 3]
```diff
@@ -83,3 +91,41 @@ var cidIdCmd = &cli.Command{
...
+
+		data, err := io.ReadAll(f)
+		if err != nil {
+			return xerrors.Errorf("error reading car file: %w", err)
+		}
+
+		blobr := bytes.NewReader(data)
+
+		hdr, err := car.LoadCar(ctx, bs, blobr)
```
Comment (travisperson) [inline cmd/lotus-shed/cid.go:120]: No need to do a `io.ReadAll` here and then create a new `io.Reader` as `os.FIle` already implements `io.Reader` which can be passed directly into `car. LoadCar`.

**Pull request merged**